moto/batch/responses.py

```diff
@@ -0,0 +1,265 @@
+from __future__ import unicode_literals
+from moto.core.responses import BaseResponse
+from .models import batch_backends
+from six.moves.urllib.parse import urlsplit
+
+from .exceptions import AWSError
+
+import json
+
+
+class BatchResponse(BaseResponse):
+    def _error(self, code, message):
+        return json.dumps({'__type': code, 'message': message}), dict(status=400)
+
+    @property
+    def batch_backend(self):
+        """
+        :return: Batch Backend
+        :rtype: moto.batch.models.BatchBackend
+        """
+        return batch_backends[self.region]
+
+    @property
+    def json(self):
+        if self.body is None or self.body == '':
+            self._json = {}
+        elif not hasattr(self, '_json'):
+            try:
+                self._json = json.loads(self.body)
+            except json.JSONDecodeError:
+                print()
+        return self._json
+
+    def _get_param(self, param_name, if_none=None):
+        val = self.json.get(param_name)
+        if val is not None:
+            return val
+        return if_none
+
+    def _get_action(self):
+        # Return element after the /v1/*
+        return urlsplit(self.uri).path.lstrip('/').split('/')[1]
+
+    # CreateComputeEnvironment
+    def createcomputeenvironment(self):
```

Comment:
I am simply amazed that the AWS Batch team decided to allow the totally deviant naming scheme for their endpoints. CamelCase, snake_case, sure. But `POST /v1/createcomputeenvironment HTTP/1.1` ? Very confusing.

Thanks for the comments to help future developers parse these method names.

Reply:
Yeah its most annoying. But the benefit they get at least is entire teams can be isolated from one another and make apis without worrying how others are doing it... as long as they provide the spec for the client libraries everyone uses. But consistency would be nice, and then again they should just purge teh xml apis too.